### PR TITLE
Implement runner.addresses property

### LINKED ISSUE #3036: Access to socket information when using AppRunner
## Long story short

Sometimes I like to listen on port 0 to allow for any and need to access the actual port after binding.

## Expected behaviour

A public way to access socket information like the port number.

## Actual behaviour

I have to reach into the private `TCPSite._server`.

### PR DESCRIPTION
**Pull request by asvetlov**
Fix #3036

### CODE REVIEW

**Comment by asvetlov** [comment]
`IPEndpoint(ipaddress, int)` or `UNIXEndpoint(str)`?

**Comment by asvetlov** [comment]
Hmm. IPv6 has additional `flow_info` and `scope_id`.
I have not enough information how people use it but suspect keeping everything is essential.
My motivation is: let's support the same naming structures as well-known `socket` module do.
Later we can invent a better API by adding `.endpoints` property :)

File: aiohttp/web_runner.py [hunk 1]
```diff
@@ -148,12 +148,18 @@ def __init__(self, *, handle_signals=False, **kwargs):
         self._handle_signals = handle_signals
         self._kwargs = kwargs
         self._server = None
-        self._sites = set()
+        self._sites = []
```
Comment (webknjaz) [inline aiohttp/web_runner.py:151]: @asvetlov what is the motivation behind this?

Comment (asvetlov) [reply aiohttp/web_runner.py:151]: To keep a stable order of returned addresses.
Registering a new site in a runner is not a bottleneck, we can live with O(n) for it.

**Pull request merged**